hubspot/errors.go

```diff
@@ -74,6 +75,9 @@ func (c *Connector) interpretJSONError(res *http.Response, body []byte) error {
 	}
 
 	switch res.StatusCode {
+	// Hubspot sends us a 400 when the search endpoint returns over 10K records.
+	case http.StatusBadRequest:
+		return createError(ErrSearchLimitExceeded, apiError)
```

Comment:
I'm sure I know the answer to this, but is there really no other hint being included in the response? Headers with extra details? `400` (alone) is such an inappropriate marker for this scenario. I know you didn't design it, I'm just baffled that there's no way to tell the difference between "too many records found" vs. "hey user, you sent us garbage".

Reply:
Nope :( It's annoying that there's not even one way to tell if it's a legit 400 or a 400 due to the search limit. I inspected the raw error that we get and here it is -
![Screenshot 2023-12-14 at 5 22 01 PM](https://github.com/amp-labs/connectors/assets/18614743/2d65bf9e-e0ab-46fa-ac0b-51b5558692ae)
![Screenshot 2023-12-14 at 5 21 48 PM](https://github.com/amp-labs/connectors/assets/18614743/bfdb60de-60fa-4c39-91aa-a4ce8ba4026d)